typescript-json-schema.ts

```diff
@@ -1653,7 +1653,18 @@ function normalizeFileName(fn: string): string {
     return fn;
 }
 
-export function exec(filePattern: string, fullTypeName: string, args = getDefaultArgs()): void {
+function defaultCallback (err: Error | null): void {
+    if (err !== null) {
+        throw err;
+    }
+}
+
+export function exec(
+    filePattern: string,
+    fullTypeName: string,
+    args = getDefaultArgs(),
+    callback: (err: Error | null) => void = defaultCallback
```

Comment:
Let's not add a callback. Just throw an error like before.

Reply:
If you want, you can mark the function as async in a follow up pull request. I prefer promises over callbacks.

Reply:
The callback is necessary to trigger the tests in `test/out-option.test.ts`. The only alternative I see would be to use `fs.writeFileSync` and `fs.mkdirSync`. Do you want me to do so ? I'm open to suggestions if you have other ideas.

Reply:
I see. Yes, let's use the sync versions for now.

Reply:
I prefer promises to. I useda callback to minimize the change.

Let's put this PR on hold. I'll open another to use a promise, and once the promise is merged, I'll update this one.

Reply:
Sounds good. Thanks!